### PR TITLE
CHK-568: Add optional custom address prop to place details

### PR DESCRIPTION
**Pull request by wdsrocha**
#### What problem is this solving?

Sometimes we might want to render the details of places that aren't from the user, like for pickup store. Now it's also possible to use place details inside address list, if we want to.

#### How should this be manually tested?

Same as https://github.com/vtex-apps/checkout-shipping/pull/31

### CODE REVIEW

**Review by jeffersontuc — APPROVED** [review]

**Review by silviasfon — APPROVED** [review]

File: react/PlaceDetails.tsx [hunk 1]
```diff
@@ -26,8 +28,10 @@ const PlaceDetails: React.FC<Props> = ({
   display = 'extended',
   hiddenFields = [],
   onEdit,
+  customAddress,
 }) => {
-  const { address, rules } = useAddressContext()
+  const { address: contextAddress, rules } = useAddressContext()
```
Comment (RodrigoCMoraes) [inline react/PlaceDetails.tsx:33]: [@question](https://conventionalcomments.org): what is this `rules`?

File: docs/README.md [hunk 2]
```diff
@@ -22,7 +22,7 @@ Country data is sorted according to [Figma designs](https://www.figma.com/file/u
 
 ## Place Details
 
-The Place Details component displays a summary of a place according to the address it receives from the wrapper orchestrator, and the description of the structure that receives as prop. There are three variations of PlaceDetails according to the amount of information that needs to be shown in a particular scenario. Below you can see a description of them.
+The Place Details component displays a summary of a place according to the address from the wrapper orchestrator or a custom address passed via props. There are three variations of PlaceDetails according to the amount of information that needs to be shown in a particular scenario. Below you can see a description of them.
```
Comment (RodrigoCMoraes) [inline docs/README.md:25]: [@praise](https://conventionalcomments.org): good to see docs being updated!

**Comment by RodrigoCMoraes** [comment]
[@question](https://conventionalcomments.org): why we don't have some sort of tests (Unit Test, Integration Test, etc...) in this PR?

File: react/PlaceDetails.tsx (hunk 1 above)
Comment (wdsrocha) [reply react/PlaceDetails.tsx:33]: `rules` are a set of address rules and settings specific to a country ([AddressRules typing](https://github.com/vtex-apps/address-context/blob/61cbaf1ed0a7cdbba942f8cd36a46647b12cb50c/react/types.tsx#L71-L73)). For example, you can check the brazilian rules [here](https://github.com/vtex-apps/countries-data/blob/main/packages/bra/vtex.country-data-settings/configuration.json). This file includes things used for input validation (like CPF, phone, etc) and which components we should render on screen. To better visualize this, the property [display](https://github.com/vtex-apps/countries-data/blob/main/packages/bra/vtex.country-data-settings/configuration.json#L167) on extended mode describes the format that we want to show, like in the pictures bellow. For UK users, it makes no sense to display number, as it is already handled by address line 1, so we hide this from the user.

All this information comes from the `useAddressContext`

BR | UK
-|-
image | image

Obs: first picture is outdated with the current setting (there is no `receiverName` property) and UK rules haven't been created yet.

File: docs/README.md (hunk 2 above)
Comment (wdsrocha) [reply docs/README.md:25]: thanks :^D

btw, [this bot](https://github.com/vtex-apps/place-components/pull/47#issuecomment-790680631) helps a lot to keep the documentation updated by forcing the PR creator to explicitly define the status of the docs

[See more about it](https://github.com/vtex-apps/docs-bot)

File: react/PlaceDetails.tsx (hunk 1 above)
Comment (RodrigoCMoraes) [reply react/PlaceDetails.tsx:33]: Thanks for this great explanation!

**Comment by wdsrocha** [comment]
> [@question](https://conventionalcomments.org): why we don't have some sort of tests (Unit Test, Integration Test, etc...) in this PR?

I thought it wasn't necessary, but thinking about it now, it would be worth it and quick to do. But as suggested by my peer in the other repository, this feature might not be needed at all, so I'll just close this PR and remove the necessity to even create a test :)

File: docs/README.md (hunk 2 above)
Comment (RodrigoCMoraes) [reply docs/README.md:25]: Awesome! That's really good to know that we have this solution. But we have some solution for [pending docs](https://github.com/vtex-apps/io-documentation/issues?page=4&q=is%3Aissue+is%3Aopen)?

**Comment by RodrigoCMoraes** [comment]
> > [@question](https://conventionalcomments.org): why we don't have some sort of tests (Unit Test, Integration Test, etc...) in this PR?
> 
> I thought it wasn't necessary, but thinking about it now, it would be worth it and quick to do. But as suggested by my peer in the other repository, this feature might not be needed at all, so I'll just close this PR and remove the necessity to even create a test :)

Do you know if we have an initiative to add [coverage](https://martinfowler.com/bliki/TestCoverage.html) or something like this on CI? 
IMHO, we have more wins than losses adopting a [testing culture](https://martinfowler.com/articles/testing-culture.html). Do you know if we already had discussed this topic in past?

File: react/PlaceDetails.tsx (hunk 1 above)
Comment (mredigonda) [reply react/PlaceDetails.tsx:33]: Agree, awesome explanation 🙂

Maybe a good place to write down some words about what these `rules` mean could be the README of this project: https://github.com/vtex-apps/address-context

Super happy that you are taking good care of this project Wes! 😸

File: docs/README.md (hunk 2 above)
Comment (wdsrocha) [reply docs/README.md:25]: We don't. It only creates the issue and never notifies the codeowners again 😞. Maybe this bot could be improved to send a warning from time to time? Does that make sense to you?

**Comment by wdsrocha** [comment]
@RodrigoCMoraes 

> Do you know if we have an initiative to add coverage or something like this on CI?

Not _yet_, but having code coverage is definitely a desired future. I just created a ["testing improvements" doc](https://www.notion.so/vtexhandbook/Testing-Quick-Wins-Improvements-99bcd1aacebe40f48ad744546770d30f) for the Checkout Squad, which I'll add more details in how to handle this problem there.

> IMHO, we have more wins than losses adopting a testing culture. Do you know if we already had discussed this topic in past?

Yes, the front-end chapter is well aware of this and we are improving, but it's still not a strong culture and we lack proper tooling (for example, no code coverage on CI!). I believe that by improving the tooling and embracing this culture on my own team, we'll be able to extend our solutions to the whole chapter and give a good example on why the testing culture is worth it.

**Comment by wdsrocha** [comment]
Closing this since @lucasecdb [proposed a good solution that does not requires the necessity to change this repository](https://github.com/vtex-apps/checkout-shipping/pull/31#discussion_r595964795).

**Pull request closed without merging**